src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -477,8 +478,8 @@ void BaseCouplingScheme::addConvergenceMeasure(
     bool                        doesLogging)
 {
   ConvergenceMeasureContext convMeasure;
-  PRECICE_ASSERT(_allData.count(dataID) == 1, "Data with given data ID must exist!");
-  convMeasure.couplingData = _allData[dataID];
+  PRECICE_ASSERT(getAllData().count(dataID) == 1, "Data with given data ID must exist!");
+  convMeasure.couplingData = getAllData()[dataID];
```

Comment:
```suggestion
  auto allData = getAllData();
  PRECICE_ASSERT(allData.count(dataID) == 1, "Data with given data ID must exist!");
  convMeasure.couplingData = allData[dataID];
```

or if you really want to be fancy:

```suggestion
  convMeasure.couplingData = [&] {
    auto allData = getAllData();
    PRECICE_ASSERT(allData.count(dataID) == 1, "Data with given data ID must exist!");
    return std::move(allData[dataID]);
  }();
```